### ISSUE TITLE
corner case in save_results and load_results

### ISSUE THREAD

**Issue by quaquel** [quaquel/EMAworkbench#90]
if outcome names contain a ',' this breaks the outcomes metadata and thus breaks loading it again after storing.

Best solution is to explicitly wrap the metadata fields within string quotes, either directly, or by turning metadata into a data frame first and rely on pandas for generating and reading the csv. However, this is a backward incompatible change and is thus to be handled in the 2.1 branch.

```python
# in save_results
outcomes_meta = {k:str(outcomes[k].shape) for k in outcomes.keys()}
outcomes_meta = pd.DataFrame.from_dict(outcomes_meta, orient='index')
outcomes_meta = outcomes_meta.to_csv(header='False', index='False')

# possibly include columns 'name', 'shape' and possibly dtype?

# in load results
outcomes_meta = z.extractfile('outcomes metadata.csv')
outcomes_meta = pd.read_csv(outcomes_meta, header=None, inde_col=False)
```

**Comment by jasonrwang** [comment]
Could this potentially also be resolved by pickling the results instead? Then for backwards compatibility, an `if` statement can check the filetype that the user wants to load?

Someone wrote a [little article](http://www.benfrederickson.com/dont-pickle-your-data/) about why not to use it. I'm not very familiar with the backend, so maybe there's a good reason not to.

**Comment by quaquel** [comment]
I have fixed this in the 2.1 branch. Some minor work is still needed in the analysis branch before 2.1 is ready for official release.

**Comment by jasonrwang** [comment]
I read the original post again and see the metadata --> df part more clearly. Thanks!
